sql/core/src/test/resources/sql-tests/inputs/group-analytics.sql

```diff
@@ -18,9 +18,13 @@ AS courseSales(course, year, earnings);
 
 -- ROLLUP
 SELECT course, year, SUM(earnings) FROM courseSales GROUP BY ROLLUP(course, year) ORDER BY course, year;
+SELECT course, year, SUM(earnings) FROM courseSales GROUP BY ROLLUP (course, year) ORDER BY course, year;
+SELECT course, year, SUM(earnings) FROM courseSales GROUP BY ROLLUP(course, year, (course, year)) ORDER BY course, year;
 
 -- CUBE
 SELECT course, year, SUM(earnings) FROM courseSales GROUP BY CUBE(course, year) ORDER BY course, year;
+SELECT course, year, SUM(earnings) FROM courseSales GROUP BY CUBE (course, year) ORDER BY course, year;
+SELECT course, year, SUM(earnings) FROM courseSales GROUP BY CUBE(course, year, (course, year)) ORDER BY course, year;
```

Comment:
Could you add more test patterns? For example,
```
scala> sql("select a, b, sum(c) from t group by cube ((a, b), ())").show()
+---+---+------+
|  a|  b|sum(c)|
+---+---+------+
+---+---+------+

postgres=# select a, b, sum(c) from t group by cube ((a, b), ());
2020-11-27 15:43:52.365 JST [23118] ERROR:  syntax error at or near ")" at character 52
2020-11-27 15:43:52.365 JST [23118] STATEMENT:  select a, b, sum(c) from t group by cube ((a, b), ());
ERROR:  syntax error at or near ")"
LINE 1: select a, b, sum(c) from t group by cube ((a, b), ());
```

Reply:
Yea, PostgresSQL not support empty set.
Will add more test after code refactor

Reply:
Btw, any database-like system supporting an empty set in CUBE?

Reply:
> Btw, any database-like system supporting an empty set in CUBE?

Have not see this point in their doc, and I have no env to test for this...

Reply:
```
// PostgreSQL
postgres=# select a, b, sum(c) from t group by grouping sets((a, b), (a), ());
 a | b | sum 
---+---+-----
   |   |   3
 1 | 1 |   1
 2 | 2 |   2
 2 |   |   2
 1 |   |   1
(5 rows)

postgres=# select a, b, sum(c) from t group by cube((a, b), (a), ());
ERROR:  syntax error at or near ")"
LINE 1: select a, b, sum(c) from t group by cube((a, b), (a), ());

// Spark
scala> sql("select a, b, sum(c) from t group by grouping sets((a, b), (a), ());").show()
+----+----+------+
|   a|   b|sum(c)|
+----+----+------+
|   1|   1|     1|
|   2|   2|     2|
|null|null|     3|
|   1|null|     1|
|   2|null|     2|
+----+----+------+

scala> sql("select a, b, sum(c) from t group by cube((a, b), (a), ());").show()
+----+----+------+
|   a|   b|sum(c)|
+----+----+------+
|   1|   1|     1|
|   1|null|     1|
|null|null|     3|
|   2|   2|     2|
|null|null|     3|
|   1|null|     1|
|   1|   1|     1|
|   1|   1|     1|
|   2|   2|     2|
|   2|null|     2|
|   2|null|     2|
|   2|   2|     2|
|   2|   2|     2|
|   1|   1|     1|
+----+----+------+

Reply:
May be we should add a check rule that cube/rollup groupingsets should not have empty set?